### PR TITLE
Groups gatherer

### PR DESCRIPTION
**Pull request by CDimonaco**
Groups retrieving gatherer, reads and parse the groups file, default `/etc/group`

### CODE REVIEW

**Review by arbulu89 — APPROVED** [review]
Noice!!
Some nitpicks. Feel free to merge.
PD: if you change the format (`id` to `gid` remember to change wanda docs)

File: internal/factsengine/gatherers/groups.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+	GroupsFileDecodingError = entities.FactGatheringError{
+		Type:    "groups-decoding-error",
+		Message: "error deconding groups file",
+	}
+)
+
+type GroupsEntry struct {
+	Name  string   `json:"name"`
+	ID    uint64   `json:"id"`
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply internal/factsengine/gatherers/groups.go:36]: yes!

File: internal/factsengine/gatherers/groups.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		return nil, err
+	}
+
+	var unmarshalled []interface{}
+	err = json.Unmarshal(marshalled, &unmarshalled)
+	if err != nil {
+		return nil, err
+	}
+
+	return entities.NewFactValue(unmarshalled, entities.WithStringConversion())
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply internal/factsengine/gatherers/groups.go:134]: you're right, done

File: internal/factsengine/gatherers/groups.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+func NewDefaultGroupsGatherer() *GroupsGatherer {
+	return NewGroupsGatherer(GroupsFilePath)
+}
+
+func NewGroupsGatherer(groupsFilePath string) *GroupsGatherer {
+	return &GroupsGatherer{groupsFilePath: groupsFilePath}
+}
+
+func (g *GroupsGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply internal/factsengine/gatherers/groups.go:53]: done

File: internal/factsengine/gatherers/groups.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	factValues, err := mapGroupsEntriesToFactValue(entries)
+	if err != nil {
+		return nil, GroupsFileDecodingError.Wrap(err.Error())
+	}
+
+	for _, requestedFact := range factsRequests {
+		facts = append(facts, entities.NewFactGatheredWithRequest(requestedFact, factValues))
+	}
+
+	log.Infof("requested %s facts gathered", GroupsGathererName)
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply internal/factsengine/gatherers/groups.go:81]: done

File: internal/factsengine/gatherers/gatherer.go [hunk 5]
```diff
@@ -10,6 +10,7 @@ type FactGatherer interface {
 
 func StandardGatherers() map[string]FactGatherer {
 	return map[string]FactGatherer{
+		GroupsGathererName:          NewDefaultGroupsGatherer(),
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply internal/factsengine/gatherers/gatherer.go:13]: all ordered!

**Comment by CDimonaco** [comment]
All done, wanda docs updated in the dedicated pr

**Pull request merged**